docs/src/main/asciidoc/security-protect-a-service-application-by-using-oidc-bearer-authentication-tutorial.adoc

```diff
@@ -0,0 +1,327 @@
+
+First compile it:
+
+include::{includes}/devtools/build.adoc[]
+
+Then run it:
+
+[source,bash]
+----
+java -jar target/quarkus-app/quarkus-run.jar
+----
+
+=== Run the application in Native mode
+
+This same demo can be compiled into native code: no modifications required.
+
+This implies that you no longer need to install a JVM on your
+production environment, as the runtime technology is included in
+the produced binary, and optimized to run with minimal resource overhead.
+
+Compilation will take a bit longer, so this step is disabled by default;
+let's build again by enabling the `native` profile:
+
+include::{includes}/devtools/build-native.adoc[]
+
+After getting a cup of coffee, you'll be able to run this binary directly:
+
+[source,bash]
+----
+./target/security-openid-connect-quickstart-1.0.0-SNAPSHOT-runner
+----
+
+=== Test the application
+
+See <<keycloak-dev-mode, Running the Application in Dev mode>> section above about testing your application in a dev mode.
+
+You can test the application launched in JVM or Native modes with `curl`.
+
+The application is using Bearer authentication and the first thing to do is obtain an access token from the Keycloak Server in
+order to access the application resources:
+
+[source,bash]
+----
+export access_token=$(\
+    curl --insecure -X POST http://localhost:8180/realms/quarkus/protocol/openid-connect/token \
+    --user backend-service:secret \
+    -H 'content-type: application/x-www-form-urlencoded' \
+    -d 'username=alice&password=alice&grant_type=password' | jq --raw-output '.access_token' \
+ )
+----
+
+The example above obtains an access token for user `alice`.
+
+Any user is allowed to access the
+`http://localhost:8080/api/users/me` endpoint
+which basically returns a JSON payload with details about the user.
+
+[source,bash]
+----
+curl -v -X GET \
+  http://localhost:8080/api/users/me \
+  -H "Authorization: Bearer "$access_token
+----
+
+The `http://localhost:8080/api/admin` endpoint can only be accessed by users with the `admin` role. If you try to access this endpoint with the
+ previously issued access token, you should get a `403` response
+ from the server.
+
+[source,bash]
+----
+curl -v -X GET \
+   http://localhost:8080/api/admin \
+   -H "Authorization: Bearer "$access_token
+----
+
+In order to access the admin endpoint you should obtain a token for the `admin` user:
+
+[source,bash]
+----
+export access_token=$(\
+    curl --insecure -X POST http://localhost:8180/realms/quarkus/protocol/openid-connect/token \
+    --user backend-service:secret \
+    -H 'content-type: application/x-www-form-urlencoded' \
+    -d 'username=admin&password=admin&grant_type=password' | jq --raw-output '.access_token' \
+ )
+----
+
+Please also see the <<integration-testing-keycloak-devservices, Dev Services for Keycloak>> section below about writing the integration tests which depend on `Dev Services for Keycloak`.
```

Comment:
The "section below" doesn't exist, so this might need to be reworked too.

Reply:
Updated. Thanks. Nice catch!